lib/linked_list.rb

```diff
@@ -94,21 +199,39 @@ def reverse
     # Space Complexity
     def find_middle_value
       raise NotImplementedError
+      
     end
 
     # find the nth node from the end and return its value
     # assume indexing starts at 0 while counting to n
     # Time Complexity:  
     # Space Complexity
     def find_nth_from_end(n)
-      raise NotImplementedError
+      #raise NotImplementedError
+      current = @head
+      index = 0
+    
+      while current != nil && index != n
+      current = current.next
+      index += 1
+
+      if current == nil
+      return nil
+      end
+
+      new_current = @head
+      while current.next != nil
+      current = current.next
+      new_current = new_current.next
+      end
+      return new_current.data
     end

```

Comment:
A couple of things:
1.  Your indentation is off here.
2.  You are missing an `end`
3.  Even beyond that, it's not working because on empty lists or lists shorter than what's given `new_current` may end up being `nil`.